test/runnable/xdtor.d

```diff
@@ -57,12 +57,33 @@ struct MixedDtor
 struct SNoDtor {}
 class CNoDtor {}
 
+struct Bar17257(E)
+{
+    ~this() @safe @nogc nothrow
+    {
+        assert(__traits(hasMember, E, "__xdtor"));
+    }
+}
+
+struct Foo17257A
+{
+    Bar17257!Foo17257A foo;
+    ~this() @safe @nogc nothrow {}
+}
+
+struct Foo17257B
+{
+    Bar17257!Foo17257B foo;
+}
+
 static assert(!__traits(hasMember, SNoDtor, "__xdtor"));
 static assert(!__traits(hasMember, CNoDtor, "__xdtor"));
 
 void test2() @safe @nogc nothrow
 {
     FieldDtor a;
+    Foo17257A foo17257A;
+    Foo17257B foo17257B;
```

Comment:
This seems to be mixing together two test cases that have nothing to do with each other into one. Please make them separate test cases.